### DISCUSSION TITLE
Are there any solutions for outputting some middle parameters?

### DISCUSSION THREAD

**Discussion by Coder-GAN** [eradiate/eradiate#363] [Q&A] [answered]
Hi there, thanks a lot for providing such comprehensive tools for the community. 
I am wondering if Eradiate can output some middle parameters of the simulation. For a simple example, when I use a 3D-explicit model for TOC simulation without atmospheric settings, can I get the transmittance of the canopy? In other words, how much energy can reach the ground after penetrating or through the canopy?

**Comment by leroyvn** [comment, marked as answer]
Hi @Coder-GAN, Eradiate's integrator (the algorithm that computes radiance estimates in the scene) doesn't provide access to such quantities directly: the correct way to get this is to define a measure recording the incident flux at the bottom of your canopy, then to divide it by the incident flux at the top of the canopy (which you know by construction). Such a measure doesn't exist yet but we could add it fairly easily (it's a couple of days of work, I think). There would most likely also be a few details to pay attention to when assembling the scene (mesh quality and illuminant selection) in order to get correct results.

**Reply by leroyvn** [reply]
Do not hesitate to get in touch with us at news@eradiate.eu to further discuss the use case, we'd be very happy to provide this feature it a near future.
